front/components/agent_builder/AgentBuilder.tsx

```diff
@@ -37,31 +60,57 @@ export default function AgentBuilder() {
     ? Math.min(10, EXTENDED_MAX_STEPS_USE_PER_RUN_LIMIT)
     : Math.min(10, MAX_STEPS_USE_PER_RUN_LIMIT);
 
+  const defaultValues = useMemo((): AgentBuilderFormData => {
+    if (agentConfiguration) {
+      const result = transformAgentConfigurationToFormData(
+        agentConfiguration,
+        user,
+        [user] // Always use current user as fallback - editors will be updated reactively
+      );
+
+      if (result.isOk()) {
+        return result.value;
+      } else {
+        // Handle error case - log error and return default values
+        console.error(
+          "Failed to transform agent configuration to form data:",
+          result.error
+        );
+        return getDefaultAgentFormData(user, defaultMaxSteps);
+      }
+    }
+    return getDefaultAgentFormData(user, defaultMaxSteps);
+  }, [agentConfiguration, user, defaultMaxSteps]);
+
+  // Create values object that includes async data (actions and editors)
+  const formValues = useMemo((): AgentBuilderFormData | undefined => {
+    const hasActions = actions && actions.length > 0;
+    const hasEditors = editors && editors.length > 0;
+
+    if (!hasActions && !hasEditors) {
+      return undefined; // Let defaultValues handle initial state
+    }
+
+    const updatedValues = { ...defaultValues };
+
+    if (hasActions) {
+      updatedValues.actions = actions;
+    }
+
+    if (hasEditors) {
+      updatedValues.agentSettings = {
+        ...updatedValues.agentSettings,
+        editors,
+      };
+    }
+
+    return updatedValues;
+  }, [defaultValues, actions, editors]);
+
   const form = useForm<AgentBuilderFormData>({
     resolver: zodResolver(agentBuilderFormSchema),
-    defaultValues: {
-      agentSettings: {
-        name: "",
-        description: "",
-        pictureUrl: "",
-        scope: "hidden",
-        editors: [user],
-        slackProvider: null,
-        slackChannels: [],
-        tags: [],
-      },
-      instructions: "",
-      generationSettings: {
-        modelSettings: {
-          modelId: GPT_4O_MODEL_CONFIG.modelId,
-          providerId: "openai",
-        },
-        temperature: 0.7,
-        reasoningEffort: GPT_4O_MODEL_CONFIG.defaultReasoningEffort,
-      },
-      actions: [],
-      maxStepsPerRun: defaultMaxSteps,
-    },
+    defaultValues,
+    values: formValues, // Reactive updates when actions are loaded
   });
 
   useEffect(() => {
```

Comment:
OOC: Can't that part be put in the `const formValues = useMemo` ?

Reply:
Defs